### ISSUE TITLE
FoxxManager.install fails

### ISSUE THREAD

**Issue by mpoeter** [arangodb/arangodb#4265]
## my environment running ArangoDB
I'm using the latest ArangoDB of the respective release series:
- [x] 3.3

Mode:
- [x] Single-Server

On this operating system:
- [x] Linux 
  - [x] Ubuntu .deb

### Foxx
We are trying to install a Foxx service as part of the DB initialization step in Docker. We have an init.js file with the following content:
```JavaScript
const foxx = require("@arangodb/foxx/manager")
foxx.install("git:Lean5/arango-prometheus-exporter:master", "/metrics")
```

This worked fine in ArangoDB v3.1.8. When we updated to v3.3 we noticed that this no longer works. A quick test showed that this problem already existed in 3.2 - just execute these two lines in `arangosh`. This results in the following error:
```
JavaScript exception in file '/usr/share/arangodb3/js/client/modules/@arangodb/arangosh.js' at 98,7: ArangoError 3101: syntax error in module
File: /var/lib/arangodb3-apps/_db/_system/metrics/APP/index.js
!      throw error;
!      ^
stacktrace: ArangoError: syntax error in module
File: /var/lib/arangodb3-apps/_db/_system/metrics/APP/index.js
    at Object.exports.checkRequestResult (/usr/share/arangodb3/js/client/modules/@arangodb/arangosh.js:96:21)
    at Object.install (/usr/share/arangodb3/js/client/modules/@arangodb/foxx/manager.js:281:12)
    at <shell command>:1:6
```
What I found strange is the path `APP/index.js`, because the service has no `injex.js` file in the root directory. I tried to hunt this down in the code and found that by specifying `force: true` the created files are not deleted. Executing the install function with "force" leaves two files in `/var/lib/arangodb3-apps/_db/_system/metrics/APP/`:
  * index.js - with the content: `https://github.com/Lean5/arango-prometheus-exporter/archive/master.zip`
  * manifest.json - with the content: `{ "main": "index.js" }`

Based on this I figured that `_prepareService` takes the path where `serviceInfo` is a `Buffer` instance and therefore does not correctly download the zip-File.

So the root cause seems to be either in the code that issues the POST request to `/_api/foxx` (missing content-type?) or in the according request handler.

**Comment by voondo** [comment]
Same problem here.

**Comment by mpoeter** [comment]
*push*
I was hoping that with all the provided details it should be relatively easy to find and fix this issue.

**Comment by pluma** [comment]
Hi @mpoeter and sorry about the delay. Your issue didn't get labelled so it dropped below the radar.

We no longer officially support the FoxxManager JS API in ArangoDB 3.x (and it should no longer be mentioned in the documentation), however it is still available for internal reasons.

As you have noticed, the `git:` URL scheme is no longer supported. Regular `http:` and `https:` URLs however should still work in 3.2 and later.

Try passing the URL directly:

```js
FoxxManager.install(
  'https://github.com/Lean5/arango-prometheus-exporter/archive/master.zip',
  '/metrics'
);
```

As a long term solution I strongly recommend using the HTTP API for managing Foxx services instead, which should work with e.g. curl:

https://docs.arangodb.com/3.2/HTTP/Foxx/

**Comment by pluma** [comment]
Hi again, I just checked the example and it looks like `FoxxManager.install` no longer supports URLs in 3.2, just local file paths. So you may have to download the file first and use a local path instead. Alternatively you can pass the URL to the Foxx HTTP API as `source` using a tool like curl.

**Comment by pluma** [comment]
After looking further into this it looks like there's actually a bug in the compatibility layer we added, so this might get fixed soon-ish. However I would still recommend against relying on the ArangoDB internals and suggest using the HTTP API instead.

**Comment by pluma** [comment]
If you want to follow the progress of the related bugfix, check PR #4449 and its siblings.

**Comment by mpoeter** [comment]
Hi Alan, thanks for following up on this.
I was looking for an easy way to install a Foxx service in ArangoDB Shell - something as simple as `db._createDatabase("myapp")` or `users.save('JohnSmith', 'mypassword')`, and FoxxManager seamed to be just that.
But if I understand correctly, there is no such (official) high level API and we should use the HTTP API instead. That's too bad, I found the the FoxxManager to be quite handy!

**Comment by pluma** [comment]
Yes, you should be able to use FoxxManager again in the release after the PR lands, but we don't officially provide a JS API for managing Foxx services in ArangoDB 3.x (but likely won't remove it for now).

We plan to clean up the existing API and make it official in ArangoDB 4 but that might include backwards-incompatible changes.

**Comment by OmarAyo** [comment]
Hi @mpoeter ,

Thanks for reporting this bug

We have released 3.3.4, and it includes a fix for it
I am closing this ticket now. Please feel free to reopen if the problem persists

**Comment by mpoeter** [comment]
We updated to 3.3.4 and everything works again as expected. Thanks for fixing this!
